android/src/main/java/com/oney/WebRTCModule/CameraCaptureController.java

```diff
@@ -27,18 +31,26 @@ public class CameraCaptureController extends AbstractVideoCaptureController {
     private static final String TAG = CameraCaptureController.class.getSimpleName();
 
     private boolean isFrontFacing;
+    @Nullable
+    private String currentDeviceId = null;
 
     private final Context context;
     private final CameraEnumerator cameraEnumerator;
-    private final ReadableMap constraints;
+    private ReadableMap constraints;
 
     /**
      * The {@link CameraEventsHandler} used with
-     * {@link CameraEnumerator#createCapturer}. Cached because the
-     * implementation does not do anything but logging unspecific to the camera
-     * device's name anyway.
+     * {@link CameraEnumerator#createCapturer}.
      */
-    private final CameraEventsHandler cameraEventsHandler = new CameraEventsHandler();
+    private final CameraEventsHandler cameraEventsHandler = new CameraEventsHandler() {
+
+        @Override
+        public void onCameraOpening(String cameraName) {
+            super.onCameraOpening(cameraName);
+            updateActualSize(cameraName, videoCapturer);
+            CameraCaptureController.this.currentDeviceId = findDeviceId(cameraName);
```

Comment:
You also need to find the device ID in `updateActualSize`, do you reckon we could rework this a bit to avoid doing it twice? Maybe first get the ID, then update the size with the ID?